tests/core/v5_1/test_v51_rpc_handlers.py

```diff
@@ -655,3 +652,106 @@ async def test_v51_rpc_sendFoundNodes_web3(
 
         assert second_receive.message.total == second_response.value
         assert encode_hex(second_receive.message.request_id) == request_id
+
+
+@pytest.mark.trio
+async def test_v51_rpc_recursiveFindNodes(tester, bob, make_request):
+    async with AsyncExitStack() as stack:
+        await stack.enter_async_context(bob.network())
+        bootnodes = collections.deque((bob.enr,), maxlen=4)
+        nodes = [bob]
+        for _ in range(20):
+            node = tester.node()
+            nodes.append(node)
+            await stack.enter_async_context(node.network(bootnodes=bootnodes))
+            bootnodes.append(node.enr)
+
+        # give the the network some time to interconnect.
+        with trio.fail_after(5):
+            for _ in range(1000):
+                await trio.lowlevel.checkpoint()
+
+        await make_request("discv5_bond", [bob.node_id.hex()])
+
+        target_node_id = secrets.token_bytes(32)
+        node_ids_by_distance = tuple(
+            sorted(
+                tuple(node.enr.node_id for node in nodes),
+                key=lambda node_id: compute_log_distance(target_node_id, node_id),
+            )
+        )
+        best_node_ids_by_distance = set(node_ids_by_distance[:3])
+
+        with trio.fail_after(10):
+            found_enrs = await make_request(
+                "discv5_recursiveFindNodes", [target_node_id.hex()]
+            )
+        found_node_ids = tuple(
+            ENR.from_repr(enr_repr).node_id for enr_repr in found_enrs
+        )
+
+        # Ensure that one of the three closest node ids was in the returned node ids
+        assert best_node_ids_by_distance.intersection(found_node_ids)
```

Comment:
I would de-flaky this test by removing almost the entire body of this and simply testing that it returns ENR records.  We don't need to test the RFN logic in this test, just that the request/response pattern works as expected.